c++/src/kj/compat/http.c++

```diff
@@ -3627,6 +3959,23 @@ public:
     connectionHeaders[HttpHeaders::BuiltinIndices::SEC_WEBSOCKET_VERSION] = "13";
     connectionHeaders[HttpHeaders::BuiltinIndices::SEC_WEBSOCKET_KEY] = keyBase64;
 
+    kj::Maybe<kj::String> offeredExtensions;
+    kj::Maybe<CompressionParameters> clientOffer;
+    KJ_IF_MAYBE(value, headers.get(HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS)) {
+      // Strip all `Sec-WebSocket-Extensions` except for `permessage-deflate`.
+      kj::StringPtr permittedExtension = "permessage-deflate"_kj;
+      auto extensions = _::splitPartsWithCondition(*value, ',', permittedExtension);
+
+      // Construct our final extensions string.
+      if (extensions.size() > 0) {
+        clientOffer = _::retainClientCompressorConfig(extensions.front());
+        offeredExtensions = kj::strArray(extensions, ", ");
+      }
+      KJ_IF_MAYBE(extensions, offeredExtensions) {
+        connectionHeaders[HttpHeaders::BuiltinIndices::SEC_WEBSOCKET_EXTENSIONS] = *extensions;
```

Comment:
https://github.com/capnproto/capnproto/pull/1480/commits/74e9db442bbddef576cac1139dd3eb32d82a78ef

Whoops, this ended up here because we switched from a for-loop to `kj::strArray`. I think `.emplace()` will help me solve another issue I've been trying to figure out so thanks for mentioning it!